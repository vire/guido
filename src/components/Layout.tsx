import Head from 'next/head'
import * as React from 'react'

const Layout = ({ children }: { children: string | JSX.Element[] }) => (
  <div className="c-text">
    <Head>
      <title>guido</title>
      <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Roboto+Mono|Source+Sans+Pro" />
      <link rel="stylesheet" href="static/blaze.min.css" />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      {/* favicon stuff */}
      <link rel="apple-touch-icon" sizes="180x180" href="static/apple-touch-icon.png" />
      <link rel="icon" type="image/png" sizes="32x32" href="static/favicon-32x32.png" />
      <link rel="icon" type="image/png" sizes="16x16" href="static/favicon-16x16.png" />
      <link rel="manifest" href="static/manifest.json" />
      <link rel="mask-icon" href="static/safari-pinned-tab.svg" color="#5bbad5" />
      <meta name="theme-color" content="#ffffff" />
    </Head>
    <header className="Layout__header">
      <img src="/static/header-logo" />
    </header>
    <main className="o-container o-container--large">{children}</main>
    <footer className="Layout__footer u-centered">
      <span>Â© Guido 2017 - <a href="https://github.com/vire/guido" rel='nofollow' >github/vire/guido</a></span>
    </footer>
    <style jsx global>{`
      body {
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
          'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
      }

      .Layout__header {
        height: 5vh;
        background: black;
        color: white;
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .Layout__header img {
        height: 30px;
        border: 1px solid #ffffff8f;
      }

      .Layout__footer {
        height: 5vh;
        background: black;
        color: white;
        font-size: 0.7rem;
        color: white;
      }
    `}</style>
  </div>
)

export default Layout
