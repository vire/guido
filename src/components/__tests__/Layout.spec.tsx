import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as React from 'react'
import Layout from '../Layout'

// TODO move to setup
configure({ adapter: new Adapter() })

describe('Render', () => {
  it('?', () => {
    expect(shallow(<Layout children={'foo'} />)).toMatchSnapshot()
  })
})
