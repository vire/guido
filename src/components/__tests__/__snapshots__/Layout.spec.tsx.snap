// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ? 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout>
    foo
  </Layout>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideEffect(Head)>
          <title>
            guido
          </title>
          <link
            href="//fonts.googleapis.com/css?family=Roboto+Mono|Source+Sans+Pro"
            rel="stylesheet"
          />
          <link
            href="static/blaze.min.css"
            rel="stylesheet"
          />
          <meta
            content="initial-scale=1.0, width=device-width"
            name="viewport"
          />
          <link
            href="static/apple-touch-icon.png"
            rel="apple-touch-icon"
            sizes="180x180"
          />
          <link
            href="static/favicon-32x32.png"
            rel="icon"
            sizes="32x32"
            type="image/png"
          />
          <link
            href="static/favicon-16x16.png"
            rel="icon"
            sizes="16x16"
            type="image/png"
          />
          <link
            href="static/manifest.json"
            rel="manifest"
          />
          <link
            color="#5bbad5"
            href="static/safari-pinned-tab.svg"
            rel="mask-icon"
          />
          <meta
            content="#ffffff"
            name="theme-color"
          />
        </SideEffect(Head)>,
        <header
          className="Layout__header"
        >
          <img
            src="/static/header-logo"
          />
        </header>,
        <main
          className="o-container o-container--large"
        >
          foo
        </main>,
        <footer
          className="Layout__footer u-centered"
        >
          <span>
            © Guido 2017 - 
            <a
              href="https://github.com/vire/guido"
              rel="nofollow"
            >
              github/vire/guido
            </a>
          </span>
        </footer>,
        <style
          global={true}
          jsx={true}
        >
          
      body {
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
          'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
      }

      .Layout__header {
        height: 5vh;
        background: black;
        color: white;
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .Layout__header img {
        height: 30px;
        border: 1px solid #ffffff8f;
      }

      .Layout__footer {
        height: 5vh;
        background: black;
        color: white;
        font-size: 0.7rem;
        color: white;
      }
    
        </style>,
      ],
      "className": "c-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <title>
              guido
            </title>,
            <link
              href="//fonts.googleapis.com/css?family=Roboto+Mono|Source+Sans+Pro"
              rel="stylesheet"
            />,
            <link
              href="static/blaze.min.css"
              rel="stylesheet"
            />,
            <meta
              content="initial-scale=1.0, width=device-width"
              name="viewport"
            />,
            <link
              href="static/apple-touch-icon.png"
              rel="apple-touch-icon"
              sizes="180x180"
            />,
            <link
              href="static/favicon-32x32.png"
              rel="icon"
              sizes="32x32"
              type="image/png"
            />,
            <link
              href="static/favicon-16x16.png"
              rel="icon"
              sizes="16x16"
              type="image/png"
            />,
            <link
              href="static/manifest.json"
              rel="manifest"
            />,
            <link
              color="#5bbad5"
              href="static/safari-pinned-tab.svg"
              rel="mask-icon"
            />,
            <meta
              content="#ffffff"
              name="theme-color"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "guido",
            },
            "ref": null,
            "rendered": "guido",
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "//fonts.googleapis.com/css?family=Roboto+Mono|Source+Sans+Pro",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "static/blaze.min.css",
              "rel": "stylesheet",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "initial-scale=1.0, width=device-width",
              "name": "viewport",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "static/apple-touch-icon.png",
              "rel": "apple-touch-icon",
              "sizes": "180x180",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "static/favicon-32x32.png",
              "rel": "icon",
              "sizes": "32x32",
              "type": "image/png",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "static/favicon-16x16.png",
              "rel": "icon",
              "sizes": "16x16",
              "type": "image/png",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "static/manifest.json",
              "rel": "manifest",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "color": "#5bbad5",
              "href": "static/safari-pinned-tab.svg",
              "rel": "mask-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "#ffffff",
              "name": "theme-color",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            src="/static/header-logo"
          />,
          "className": "Layout__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "/static/header-logo",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "foo",
          "className": "o-container o-container--large",
        },
        "ref": null,
        "rendered": "foo",
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            © Guido 2017 - 
            <a
              href="https://github.com/vire/guido"
              rel="nofollow"
            >
              github/vire/guido
            </a>
          </span>,
          "className": "Layout__footer u-centered",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "© Guido 2017 - ",
              <a
                href="https://github.com/vire/guido"
                rel="nofollow"
              >
                github/vire/guido
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "© Guido 2017 - ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "github/vire/guido",
                "href": "https://github.com/vire/guido",
                "rel": "nofollow",
              },
              "ref": null,
              "rendered": "github/vire/guido",
              "type": "a",
            },
          ],
          "type": "span",
        },
        "type": "footer",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "
      body {
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
          'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
      }

      .Layout__header {
        height: 5vh;
        background: black;
        color: white;
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .Layout__header img {
        height: 30px;
        border: 1px solid #ffffff8f;
      }

      .Layout__footer {
        height: 5vh;
        background: black;
        color: white;
        font-size: 0.7rem;
        color: white;
      }
    ",
          "global": true,
          "jsx": true,
        },
        "ref": null,
        "rendered": "
      body {
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
          'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
      }

      .Layout__header {
        height: 5vh;
        background: black;
        color: white;
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .Layout__header img {
        height: 30px;
        border: 1px solid #ffffff8f;
      }

      .Layout__footer {
        height: 5vh;
        background: black;
        color: white;
        font-size: 0.7rem;
        color: white;
      }
    ",
        "type": "style",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideEffect(Head)>
            <title>
              guido
            </title>
            <link
              href="//fonts.googleapis.com/css?family=Roboto+Mono|Source+Sans+Pro"
              rel="stylesheet"
            />
            <link
              href="static/blaze.min.css"
              rel="stylesheet"
            />
            <meta
              content="initial-scale=1.0, width=device-width"
              name="viewport"
            />
            <link
              href="static/apple-touch-icon.png"
              rel="apple-touch-icon"
              sizes="180x180"
            />
            <link
              href="static/favicon-32x32.png"
              rel="icon"
              sizes="32x32"
              type="image/png"
            />
            <link
              href="static/favicon-16x16.png"
              rel="icon"
              sizes="16x16"
              type="image/png"
            />
            <link
              href="static/manifest.json"
              rel="manifest"
            />
            <link
              color="#5bbad5"
              href="static/safari-pinned-tab.svg"
              rel="mask-icon"
            />
            <meta
              content="#ffffff"
              name="theme-color"
            />
          </SideEffect(Head)>,
          <header
            className="Layout__header"
          >
            <img
              src="/static/header-logo"
            />
          </header>,
          <main
            className="o-container o-container--large"
          >
            foo
          </main>,
          <footer
            className="Layout__footer u-centered"
          >
            <span>
              © Guido 2017 - 
              <a
                href="https://github.com/vire/guido"
                rel="nofollow"
              >
                github/vire/guido
              </a>
            </span>
          </footer>,
          <style
            global={true}
            jsx={true}
          >
            
      body {
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
          'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
      }

      .Layout__header {
        height: 5vh;
        background: black;
        color: white;
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .Layout__header img {
        height: 30px;
        border: 1px solid #ffffff8f;
      }

      .Layout__footer {
        height: 5vh;
        background: black;
        color: white;
        font-size: 0.7rem;
        color: white;
      }
    
          </style>,
        ],
        "className": "c-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <title>
                guido
              </title>,
              <link
                href="//fonts.googleapis.com/css?family=Roboto+Mono|Source+Sans+Pro"
                rel="stylesheet"
              />,
              <link
                href="static/blaze.min.css"
                rel="stylesheet"
              />,
              <meta
                content="initial-scale=1.0, width=device-width"
                name="viewport"
              />,
              <link
                href="static/apple-touch-icon.png"
                rel="apple-touch-icon"
                sizes="180x180"
              />,
              <link
                href="static/favicon-32x32.png"
                rel="icon"
                sizes="32x32"
                type="image/png"
              />,
              <link
                href="static/favicon-16x16.png"
                rel="icon"
                sizes="16x16"
                type="image/png"
              />,
              <link
                href="static/manifest.json"
                rel="manifest"
              />,
              <link
                color="#5bbad5"
                href="static/safari-pinned-tab.svg"
                rel="mask-icon"
              />,
              <meta
                content="#ffffff"
                name="theme-color"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "guido",
              },
              "ref": null,
              "rendered": "guido",
              "type": "title",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "//fonts.googleapis.com/css?family=Roboto+Mono|Source+Sans+Pro",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "static/blaze.min.css",
                "rel": "stylesheet",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "initial-scale=1.0, width=device-width",
                "name": "viewport",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "static/apple-touch-icon.png",
                "rel": "apple-touch-icon",
                "sizes": "180x180",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "static/favicon-32x32.png",
                "rel": "icon",
                "sizes": "32x32",
                "type": "image/png",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "static/favicon-16x16.png",
                "rel": "icon",
                "sizes": "16x16",
                "type": "image/png",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "static/manifest.json",
                "rel": "manifest",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "color": "#5bbad5",
                "href": "static/safari-pinned-tab.svg",
                "rel": "mask-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "#ffffff",
                "name": "theme-color",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="/static/header-logo"
            />,
            "className": "Layout__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "/static/header-logo",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "foo",
            "className": "o-container o-container--large",
          },
          "ref": null,
          "rendered": "foo",
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              © Guido 2017 - 
              <a
                href="https://github.com/vire/guido"
                rel="nofollow"
              >
                github/vire/guido
              </a>
            </span>,
            "className": "Layout__footer u-centered",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "© Guido 2017 - ",
                <a
                  href="https://github.com/vire/guido"
                  rel="nofollow"
                >
                  github/vire/guido
                </a>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "© Guido 2017 - ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "github/vire/guido",
                  "href": "https://github.com/vire/guido",
                  "rel": "nofollow",
                },
                "ref": null,
                "rendered": "github/vire/guido",
                "type": "a",
              },
            ],
            "type": "span",
          },
          "type": "footer",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "
      body {
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
          'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
      }

      .Layout__header {
        height: 5vh;
        background: black;
        color: white;
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .Layout__header img {
        height: 30px;
        border: 1px solid #ffffff8f;
      }

      .Layout__footer {
        height: 5vh;
        background: black;
        color: white;
        font-size: 0.7rem;
        color: white;
      }
    ",
            "global": true,
            "jsx": true,
          },
          "ref": null,
          "rendered": "
      body {
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
          'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
      }

      .Layout__header {
        height: 5vh;
        background: black;
        color: white;
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .Layout__header img {
        height: 30px;
        border: 1px solid #ffffff8f;
      }

      .Layout__footer {
        height: 5vh;
        background: black;
        color: white;
        font-size: 0.7rem;
        color: white;
      }
    ",
          "type": "style",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
